coreclient/src/clients/update_key.rs

```diff
@@ -121,11 +174,27 @@ mod update_key_flow {
         ) -> anyhow::Result<Vec<ChatMessage>> {
             let Self {
                 mut group,
+                mut chat,
                 ds_timestamp,
+                own_id,
             } = self;
-            let group_messages = group
+            let (mut group_messages, group_data) = group
                 .merge_pending_commit(&mut *connection, None, ds_timestamp)
                 .await?;
+
+            if let Some(group_data) = group_data {
+                let attribute_messages = update_chat_attributes(
+                    connection,
+                    notifier,
+                    &mut chat,
+                    own_id,
+                    group_data,
+                    ds_timestamp,
+                )
+                .await?;
+                group_messages.extend(attribute_messages);
```

Comment:
I think `group_messages` should be a &mut parameter of `update_chat_attributes`. Then you can just push there without allocating a vec of 0/1/2 elements.